components/script/dom/htmlmediaelement.rs

```diff
@@ -825,6 +825,39 @@ impl HTMLMediaElement {
         }
     }
 
+    /// Queues a task to run the dedicated [steps when the media reaches the end][steps]
+    ///
+    /// [steps]: https://html.spec.whatwg.org/multipage/#reaches-the-end
+    fn queue_reaches_the_end_steps(&self) {
+        let window = window_from_node(self);
+        let this = Trusted::new(self);
+
+        let _ = window.task_manager().media_element_task_source().queue(
+            task!(reaches_the_end_steps: move || {
+                let this = this.root();
+                // Step 3.1.
+                this.upcast::<EventTarget>().fire_event(atom!("timeupdate"));
+
+                // Step 3.2.
+                if this.Ended() && !this.Loop() {
```

Comment:
I think this should be `!this.Paused()` instead of `!this.Loop()`

Reply:
You're right, I misread the steps